native/src/main/java/org/ballerinax/asb/receiver/MessageReceiver.java

```diff
@@ -295,18 +295,17 @@ public static Object complete(BObject receiverClient, BString lockToken) {
      * Abandons message & make available again for processing from Queue or Subscription, based on messageLockToken.
      *
      * @param receiverClient Ballerina ASB client object
-     * @param lockToken      Message lock token.
+     * @param message        Message object.
      * @return An error if failed to abandon the message.
      */
-    public static Object abandon(BObject receiverClient, BString lockToken) {
+    public static Object abandon(BObject receiverClient, BMap<BString, Object> message) {
         try {
             ServiceBusReceiverClient receiver = getReceiverFromBObject(receiverClient);
-            ServiceBusReceivedMessage message = (ServiceBusReceivedMessage) receiverClient
-                    .getNativeData(lockToken.getValue());
-            receiver.abandon(message);
-            receiverClient.getNativeData().remove(lockToken.getValue());
+            ServiceBusReceivedMessage nativeMessage = getNativeMessage(message);
+            ;
```

Comment:
Is this semi-colon intentional ?

Reply:
No,My Bad.. Addressed in [f421f48](https://github.com/ballerina-platform/module-ballerinax-azure-service-bus/pull/196/commits/f421f488dc38f28b2d64f54f6162afc267b93148)